internal/cli/internal/command/machine/run.go

```diff
@@ -284,47 +294,35 @@ func runMachineRun(ctx context.Context) error {
 		input.OrgSlug = org.ID
 	}
 
-	machine, _, err := client.LaunchMachine(ctx, input)
+	flapsClient, err := flaps.New(ctx, app)
 	if err != nil {
-		return fmt.Errorf("could not launch machine: %w", err)
+		return fmt.Errorf("could not make flaps client: %w", err)
 	}
 
-	if flag.GetBool(ctx, "detach") {
-		fmt.Fprintf(io.Out, "Machine: %s\n", machine.ID)
-		return nil
+	mach, err := flapsClient.Launch(ctx, input)
+	if err != nil {
+		return fmt.Errorf("could not launch machine: %w", err)
 	}
 
-	// apiClient := cmdCtx.Client.API()
-
-	opts := &logs.LogOptions{
-		AppName: app.Name,
-		VMID:    machine.ID,
+	var machineBody api.V1Machine
+	if err := json.Unmarshal(mach, &machineBody); err != nil {
+		return errors.Wrap(err, "Machine launch return value could not be parsed")
 	}
 
-	stream, err := logs.NewNatsStream(ctx, client, opts)
-
+	// wait for machine //
+	_, err = flapsClient.Wait(ctx, &machineBody)
 	if err != nil {
-		terminal.Debugf("could not connect to wireguard tunnel, err: %v\n", err)
-		terminal.Debug("Falling back to log polling...")
-
-		stream, err = logs.NewPollingStream(ctx, client, opts)
-		if err != nil {
-			return err
-		}
+		return errors.Wrap(err, "Firecracker VM failed to launch")
 	}
 
-	presenter := presenters.LogPresenter{}
+	fmt.Fprintf(io.Out, "Success! A machine has been successfully launched\n")
+	fmt.Fprintf(io.Out, " Machine ID: %s\n", machineBody.ID)
+	fmt.Fprintf(io.Out, " Instance ID: %s\n", machineBody.InstanceID)
+	fmt.Fprintf(io.Out, " State: %s\n", machineBody.State)
```

Comment:
We may want to do a `flapsClient.Get` after `Wait` completes successfully to get the most up-to-date state of the machine.